src/wormhole_transit_relay/transit_server.py

```diff
@@ -83,272 +109,96 @@ def rawDataReceived(self, data):
         # practice, this buffers about 10MB per connection, after which
         # point the sender will only transmit data as fast as the
         # receiver can handle it.
-        if self._sent_ok:
-            if not self._buddy:
-                # Our buddy disconnected (we're "jilted"), so we hung up too,
-                # but our incoming data hasn't stopped yet (it will in a
-                # moment, after our disconnect makes a roundtrip through the
-                # kernel). This probably means the file receiver hung up, and
-                # this connection is the file sender. In may-2020 this
-                # happened 11 times in 40 days.
-                return
-            self._total_sent += len(data)
-            self._buddy.transport.write(data)
-            return
-
-        # handshake is complete but not yet sent_ok
-        self.sendLine(b"impatient")
-        if self._log_requests:
-            log.msg("transit impatience failure")
-        return self.disconnect_error() # impatience yields failure
+        self._state.got_bytes(data)
 
     def _got_handshake(self, token, side):
```

Comment:
nit: does it make sense to inline this with `TransitConnection#lineReceived`?

Reply:
It could be, I suppose.

I guess I probably felt that method is already "kind of" big and calls this one in two ways: with `side==None` or not depending on "old-style" or "new-style" handshake. Do you think it would be clearer to read inlined?

Reply:
I think it does make it a little more readable. I can understand the reluctance to make `lineReceived` any bigger but it looks to me like you would just save one line here (`self.setRawMode()`) and end up duplicate the conditional logic.